docs/reference/api.md

```diff
@@ -0,0 +1,77 @@
+# API
+
+Pebble exposes an API to allow remote clients to interact with the daemon. The API can start and stop services, add configuration layers to the plan, and so on.
+
+## Accessing the API
+
+The API uses HTTP over a Unix socket, with access to the API controlled by user ID. If `pebble run` is started with the `--http <address>` option, Pebble exposes a limited set of open-access API endpoints using the given TCP address. See {ref}`api-access-levels`.
+
+See below for some examples of how to use the API. For more examples, see "How to use the Pebble API". <!-- [David] Link to the how-to guide -->
+
+There's a Go client for the API. See the [Go client documentation](https://pkg.go.dev/github.com/canonical/pebble/client) and the examples below.
+
+We try to never change the underlying HTTP API in a backwards-incompatible way. However, in rare cases we may change the Go client in a backwards-incompatible way.
+
+There's also a Python client for the API. The Python client is part of the `ops` library used by Juju charms. See the [Ops documentation](https://ops.readthedocs.io/en/latest/).
+
+### Go client examples
+
+TODO: go client code snippets.
+
+### cURL examples
+
+TODO: curl examples.
+
+## API access levels
+
+API endpoints fall into one of three access levels, from least restricted to most restricted. See {ref}`api-access-levels`.
+
+### Identities
+
+Pebble has the concept of named "identities", which lets administrators manage users' access to the API. See {ref}`controlling-api-access-using-identities`.
+
+## Common parameters
+
+### timeout
+
+The format of the timeout is a decimal number with an optional unit suffix (e.g., "300ms", "1.5s", "2h45m").
```

Comment:
Can we basically copy this from the Go docs (to mention that it can be a sequence of number+unit.

> A duration string is a possibly signed sequence of decimal numbers, each with optional fraction and a unit suffix, such as "300ms", "-1.5h" or "2h45m". Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".

Reply:
OK.

FYI, I removed the "possibly signed" part because it doesn't make sense for the timeout (or any parameter that is a duration in Pebble) to be negative. I hope this makes sense.